### DISCUSSION TITLE
Single file containing a list of items

### DISCUSSION THREAD

**Discussion by probablySophie** [toml-rs/toml#811] [Q&A] [answered]
Hi, I was hoping to read a file 

A super simple abstraction of the TOML file I'm hoping to read
```
[[item-identifier]]
key1 = date1
key2 = value2

[[item-identifier]]
key1 = date2
ley2 = value4

and so on...
```

The line of code I currently have for parsing
```rust
let Ok(extracted) = toml::from_str::<T>(&file_string)
```

Everything's fine when reading from a file containing a single item's key/value pairs *and no [[item-identifier]]*
```
key1 = date1
key2 = value2
```

Its just when I want to add identifiers that everything falls apart.

Error message
```
TOML parse error at line 1, column 1
  |
1 | [[item-identifier]]
  | ^
missing field `date`
```

I know I can just `file_contents.split("[[item-identifier]]")` but I was hoping for an idiomatic/expandable way of doing it.  
Possibly even with hopes to have multiple different types of item & identifier in the same file.

(And the reason I want to do this specifically is that I'm following some specification docs and I'd really rather not create a different file layout)

Thank you!

**Comment by epage** [comment]
`toml::from_str` is parsing the TOML into a specific schema that is defined using Rust types in serde.  I'd need a reproduction case that included both the schema and the data to be able to answer.

**Comment by probablySophie** [comment, marked as answer]
I have solutions that do exactly what past me wanted!

Reading multiple items from a file:  

Instead of trying to read a `Vec<item-identifier>`, create a parent Struct:
```rust
#[derive(Serialize, Deserialize)]
struct Parent
{
    item_identifier: Vec<item_identifier>
    // Assuming that the struct item_identifier exists elsewhere
}
```
And then just read `toml::from_str::<Parent>(&file_string)`!

Past me was also about to run head-first into `item-identifier` vs `item_identifier` (the first not being a valid name), and that can be solved with 

```rust
#[derive(Serialize, Deserialize)]
#[serde(rename_all="kebab-case")]
struct Parent
{
    items: Vec<item_identifier>
}
```
